modules/programs/go.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.go;
+
+in
+
+{
+  meta.maintainers = [ maintainers.rvolosatovs ];
+
+  options = {
+    programs.go = {
+      enable = mkEnableOption "Go";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.go;
+        defaultText = "pkgs.go";
+        description = "The Go package to use.";
+      };
+
+      packages = mkOption {
+        type = with types; attrsOf path;
+        example = literalExample ''
+          {
+            "golang.org/x/time/rate" = builtins.fetchGit "https://go.googlesource.com/text";
+          }
+        '';
+        description = "Packages to add to GOPATH.";
+      };
+
+      goPath = mkOption {
+        type = with types; nullOr str;
+        default = "go";
+        example = "";
+        description = "GOPATH relative to HOME";
+      };
+
+      goBin = mkOption {
+        type = with types; nullOr str;
+        default = "";
+        example = ".local/bin.go";
+        description = "GOBIN relative to HOME";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      home.file = let
+        goPath = if cfg ? goPath then cfg.goPath else "go";
+      in
+      mapAttrs' (name: source: nameValuePair "${goPath}/src/${name}" { inherit source; }) cfg.packages;
```

Comment:
I think you're missing some indentation here.

But I'd suggest to use `home.file` as a list and instead do something like
```nix
{
home.file =
  let
    mkSrc = n: v: {
      target = "${cfg.goPath}/src/${n}";
      source = v;
    };
  in
    mapAttrsToList mkSrc cfg.packages;
}
```

Reply:
Note, here I assume `goPath` has the type `str` instead of `nullOf str`.